toolchain/parser/parser_impl.cpp

```diff
@@ -135,9 +135,13 @@ auto ParseTree::Parser::ConsumeIf(TokenKind kind)
 }
```

Comment:
(This is as close as I can put a comment...)

On line 101: we no longer expect for `tree.node_impls_.size()` to end up being exactly `tokens.size()` -- we can now end up with more tree nodes than tokens in a valid parse -- so should we be adding some small factor there to avoid reallocation in the common case? Eg, `tokens.size() * 1.05`

Reply:
Per discussion, changed to calculate the tree size on the idea that it'll pay off.